Compile Error - TechTreeLayout.cpp:216:18: error: no match for ‘operator>’

```
[ 98%] Building CXX object CMakeFiles/freeorion.dir/UI/TechTreeLayout.cpp.o
[ 98%] Building CXX object CMakeFiles/freeorion.dir/UI/TechTreeWnd.cpp.o
[100%] Building CXX object CMakeFiles/freeorion.dir/UI/TextBrowseWnd.cpp.o
/home/andy/freeorion/UI/DesignWnd.cpp: In member function ‘void DesignWnd::MainPanel::DoLayout()’:
/home/andy/freeorion/UI/DesignWnd.cpp:4505:16: warning: structured binding declaration set but not used [-Wunused-but-set-variable]
 4505 |     const auto [cl_width, cl_height] = cl_sz;
      |                ^~~~~~~~~~~~~~~~~~~~~
[100%] Building CXX object CMakeFiles/freeorion.dir/util/DependencyVersions.cpp.o
In file included from /usr/include/boost/assert.hpp:58,
                 from /usr/include/boost/filesystem/detail/path_traits.hpp:24,
                 from /usr/include/boost/filesystem/path.hpp:21,
                 from /home/andy/freeorion/UI/../universe/../util/Pending.h:7,
                 from /home/andy/freeorion/UI/../universe/Tech.h:15,
                 from /home/andy/freeorion/UI/TechTreeLayout.cpp:3:
/home/andy/freeorion/UI/TechTreeLayout.cpp: In member function ‘void TechTreeLayout::AddNode(const std::string&, GG::X, GG::Y)’:
/home/andy/freeorion/UI/TechTreeLayout.cpp:216:18: error: no match for ‘operator>’ (operand types are ‘GG::X’ and ‘int’)
  216 |     assert(width > 0 && height > 0 && GetTech(tech));
      |            ~~~~~ ^ ~
      |            |       |
      |            GG::X   int
/home/andy/freeorion/UI/TechTreeLayout.cpp:216:32: error: no match for ‘operator>’ (operand types are ‘GG::Y’ and ‘int’)
  216 |     assert(width > 0 && height > 0 && GetTech(tech));
      |                         ~~~~~~ ^ ~
      |                         |        |
      |                         GG::Y    int
/home/andy/freeorion/UI/TechTreeLayout.cpp: In constructor ‘TechTreeLayout::Node::Node(const std::string&, GG::X, GG::Y)’:
/home/andy/freeorion/UI/TechTreeLayout.cpp:266:16: error: no match for ‘operator>’ (operand types are ‘GG::X’ and ‘int’)
  266 | { assert(width > 0 && height > 0 && GetTech(tech)); }
      |          ~~~~~ ^ ~
      |          |       |
      |          GG::X   int
/home/andy/freeorion/UI/TechTreeLayout.cpp:266:30: error: no match for ‘operator>’ (operand types are ‘GG::Y’ and ‘int’)
  266 | { assert(width > 0 && height > 0 && GetTech(tech)); }
      |                       ~~~~~~ ^ ~
      |                       |        |
      |                       GG::Y    int
make[2]: *** [CMakeFiles/freeorion.dir/build.make:972: CMakeFiles/freeorion.dir/UI/TechTreeLayout.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:611: CMakeFiles/freeorion.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

cmake output:

[andy@radiostation freeorion]$ cmake -DCMAKE_BUILD_TYPE=Debug
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


-- Build type CMAKE_BUILD_TYPE set to Debug
-- CCache version: 4.8.2
-- CCache configured for linux: /usr/bin/ccache
-- Use CCache for Unix Makefiles called with: ccache program /usr/bin/ccache 4.8.2
-- Python library version detected 3.12.1
-- Boost python version 312
-- Boost 1.73.0 found.
-- Found Boost components:
   filesystem;iostreams;locale;log;log_setup;serialization;system;thread
-- Boost version detected 1.81
-- Boost  found.
-- Found Boost components:
   python312
-- Boost 1.69.0 found.
-- Found Boost components:
   filesystem;regex;system
Writing file: util/Version.cpp
Writing file: cmake/FreeOrionVersion.cmake
Building v0.5+ build 2023-12-26.aa62a6e
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /home/andy/freeorion


Believe I have included all relevant info but am happy to further provide if necessary.

Fedora 39, x64.

--- a/UI/TechTreeLayout.cpp
+++ b/UI/TechTreeLayout.cpp
@@ -213,7 +213,7 @@ const TechTreeLayout::Node* TechTreeLayout::GetNode(const std::string & name) co
 }
 
 void TechTreeLayout::AddNode(const std::string& tech, GG::X width, GG::Y height) {
-    assert(width > X0 && height > Y0 && GetTech(tech));
+    assert(width > GG::X0 && height > GG::Y0 && GetTech(tech));
     auto node = new TechTreeLayout::Node(tech, width, height);
     //DebugLogger() << "Adding Node: " << node << " for tech " << tech;
     m_nodes.push_back(node);
@@ -263,7 +263,7 @@ TechTreeLayout::Node::Node(const std::string& tech, GG::X width, GG::Y height) :
     place_holder(false),
     m_width(Value(width)),
     m_height(Value(height))
-{ assert(width > X0 && height > Y0 && GetTech(tech)); }
+{ assert(width > GG::X0 && height > GG::Y0 && GetTech(tech)); }
 
 /**
  * recursively creates dummy nodes between parent and child

--- a/UI/TechTreeLayout.cpp
+++ b/UI/TechTreeLayout.cpp
@@ -213,7 +213,7 @@ const TechTreeLayout::Node* TechTreeLayout::GetNode(const std::string & name) co
 }
 
 void TechTreeLayout::AddNode(const std::string& tech, GG::X width, GG::Y height) {
-    assert(width > 0 && height > 0 && GetTech(tech));
+    assert(width > X0 && height > Y0 && GetTech(tech));
     auto node = new TechTreeLayout::Node(tech, width, height);
     //DebugLogger() << "Adding Node: " << node << " for tech " << tech;
     m_nodes.push_back(node);
@@ -263,7 +263,7 @@ TechTreeLayout::Node::Node(const std::string& tech, GG::X width, GG::Y height) :
     place_holder(false),
     m_width(Value(width)),
     m_height(Value(height))
-{ assert(width > 0 && height > 0 && GetTech(tech)); }
+{ assert(width > X0 && height > Y0 && GetTech(tech)); }
 
 /**
  * recursively creates dummy nodes between parent and child

